dimod/quadratic/cyqm/cyqm_template.pyx.pxi

```diff
@@ -705,6 +712,71 @@ cdef class cyQM_template(cyQMBase):
         
         self.cppqm.set_quadratic(ui, vi, bias)
 
+    def update(self, cyBQM_and_QM other):
+        # we'll need a mapping from the other's variables to ours
+        cdef vector[Py_ssize_t] mapping
+        mapping.reserve(other.num_variables())
+
+        cdef Py_ssize_t vi
+
+        # first make sure that any variables that overlap match in terms of
+        # vartype and bounds
+        for vi in range(other.num_variables()):
+            v = other.variables.at(vi)
+            if self.variables.count(v):
+                # there is a variable already
+                mapping.push_back(self.variables.index(v))
+
+                if self.cppqm.vartype(mapping[vi]) != other.data().vartype(vi):
+                    raise ValueError(f"conflicting vartypes: {v!r}")
+
+                if self.cppqm.lower_bound(mapping[vi]) != other.data().lower_bound(vi):
+                    raise ValueError(f"conflicting lower bounds: {v!r}")
+
+                if self.cppqm.upper_bound(mapping[vi]) != other.data().upper_bound(vi):
+                    raise ValueError(f"conflicting upper bounds: {v!r}")
```

Comment:
I'm parachuting into this PR, but I would naively expect an `update` method in Python to silently overwrite/upgrade target from source. Perhaps warn at most.

Reply:
Eh... we don't really allow too much manipulation of the bounds once they are set. Also there are a bunch of places that require this consistency. So IMO better to keep this behavior.